app/controllers/help_controller.rb

```diff
@@ -0,0 +1,68 @@
+class HelpController < ApplicationController
+  def index
+    @help_index = File.read(Rails.root.join("doc", "README.md"))
+
+    # Prefix Markdown links with `help/` unless they already have been
+    # See http://rubular.com/r/nwwhzH6Z8X
+    @help_index.gsub!(%r{(\]\()(?!help\/)([^\)\(]+)(\))}, '\1help/\2\3')
+  end
+
+  def show
+    @category = clean_path_info(path_params[:category])
+    @file = path_params[:file]
+
+    respond_to do |format|
+      format.any(:md, :html) do
+        # Note: We are purposefully NOT using `Rails.root.join`
+        path = File.join(Rails.root, "doc", @category, "#{@file}.md")
+
+        if File.exist?(path)
+          @markdown = File.read(path)
+          render "show.html.erb"
+        else
+          redirect_to help_path
+        end
+      end
+    end
+  end
+
+  private
+
+  def path_params
+    params.require(:category)
+    params.require(:file)
+
+    params
+  end
+
+  PATH_SEPS = Regexp.union(*[::File::SEPARATOR, ::File::ALT_SEPARATOR].compact)
+
+  # Taken from ActionDispatch::FileHandler
+  # Cleans up the path, to prevent directory traversal outside the doc folder.
+  def clean_path_info(path_info)
```

Comment:
I added a test to see what happens if you try to reach the README.md in the root of the project :-)